Tools/YamlCreate.ps1

```diff
@@ -31,232 +31,663 @@ filter TrimString {
     $_.Trim()
 }
 
-##########################################
-#region checksum
-while ([string]::IsNullOrWhiteSpace($URL)) {
-    $URL = Read-Host -Prompt 'Enter the URL to the installer' | TrimString
-}
-Write-Host $NewLine
-Write-Host "Downloading URL. This will take awhile..." -ForegroundColor Blue
-$WebClient = New-Object System.Net.WebClient
-
-try {
-    $Stream = $WebClient.OpenRead($URL)
-    $Hash = (Get-FileHash -InputStream $Stream -Algorithm SHA256).Hash
-}
-catch {
-    Write-Host "Error downloading file. Please run the script again." -ForegroundColor Red
-    exit 1
-}
-finally {
-    $Stream.Close()
+while ([string]::IsNullOrWhiteSpace($OptionMenu)) {
+    Clear-Host
+    Write-Host -ForegroundColor 'Cyan' -Object 'Select Mode'
+    Write-Host -ForegroundColor 'DarkCyan' -NoNewline "`n["; Write-Host -NoNewline '1'; Write-Host -ForegroundColor DarkCyan -NoNewline "]"; `
+        Write-Host -ForegroundColor 'DarkCyan' -Object ' New Manifest'
+    Write-Host -ForegroundColor 'DarkCyan' -NoNewline "`n["; Write-Host -NoNewline '2'; Write-Host -ForegroundColor DarkCyan -NoNewline "]"; `
+        Write-Host -ForegroundColor 'DarkCyan' -Object ' Update Manifest'
+    Write-Host -ForegroundColor 'DarkCyan' -NoNewline "`n["; Write-Host -NoNewline '3'; Write-Host -ForegroundColor DarkCyan -NoNewline "]"; `
+        Write-Host -ForegroundColor 'DarkCyan' -Object ' New Locale'
+    Write-Host -ForegroundColor 'DarkCyan' -NoNewline "`n["; Write-Host -NoNewline 'q'; Write-Host -ForegroundColor DarkCyan -NoNewline "]"; `
+        Write-Host -ForegroundColor 'Red' -Object ' Any key to quit'
+    $OptionMenu = Read-Host "`nSelection"
+    switch ($OptionMenu) {
+        '1' {$Option = 'New'}
+        '2' {$Option = 'Update'}
+        '3' {$Option = 'NewLocale'}
+        default {exit}
+    }
 }
 
-Write-Host "Url: $URL"
-Write-Host "Sha256: $Hash"
+##########################################
+#region checksum
 
-Write-Host $NewLine
-Write-Host "File downloaded. Please Fill out required fields."
 
 #endregion
 ##########################################
 
 ##########################################
 #region Read in metadata
 
-while ($ID.Length -lt 4 -or $ID.Length -ge 255) {
-    Write-Host 'Enter the package Id, in the following format <Publisher.Appname>'
-    $ID = Read-Host -Prompt 'For example: Microsoft.Excel' | TrimString
-}
-
-$host.UI.RawUI.ForegroundColor = "White"
 while ([string]::IsNullOrWhiteSpace($Publisher) -or $Publisher.Length -ge 128) {
-    $Publisher = Read-Host -Prompt 'Enter the publisher' | TrimString
+    Write-Host ''
+    Write-Host -ForegroundColor 'Green' -Object '[Required] Enter the publisher. For example: Microsoft'
+    $Publisher = Read-Host -Prompt 'Publisher' | TrimString
```

Comment:
You shouldn't need the Publisher or PackageName yet (see below comments as to why)

Reply:
Yep yep i realized that yesterday when i remember we should use Microsoft Corporation as the Publisher and actually Microsoft Teams as PackageName.. However i don't think it will go trough if the PackageName don't match the Publisher.PackageName? Atleast we had these problems in the early WinGet days where Publisher.PackageName had to match the Publisher and PackageName.

I've added this code back to solve this
```
while ($PackageIdentifier.Length -lt 4 -or $ID.Length -ge 255) {
    Write-Host ''
    Write-Host -ForegroundColor 'Red' -Object ' [Required] Enter the Package Identifier, in the following format <Publisher Shortname.PackageName Shortname>. For example: Microsoft.Excel'
    $PackageIdentifier = Read-Host -Prompt 'PackageIdentifier' | TrimString
}
```

Reply:
Easier then expected..
`$PackageIdentifierFolder = $PackageIdentifier.Replace('.','\')`